e2etest/utils.go

```diff
@@ -76,3 +84,150 @@ func tweakRCPort() func(rc *v1.ReplicationController) {
 	}
 	return nil
 }
+
+// Increment ip and test against CIDR after to avoid overflows.
+func incrementIP(ip string, network string) (string, error) {
+	_, ipNet, err := net.ParseCIDR(network)
+	if err != nil {
+		return "", fmt.Errorf("Failed to parse network while incrementing IP %s", err)
+	}
+
+	nextIP := net.ParseIP(ip)
+	if nextIP == nil {
+		return "", fmt.Errorf("Failed to parse IP while incrementing IP")
+	}
+
+	for i := len(nextIP) - 1; i >= 0; i-- {
+		nextIP[i]++
+		if nextIP[i] != 0 {
+			break
+		}
+	}
+
+	if !ipNet.Contains(nextIP) {
+		return "", fmt.Errorf("Overflowed CIDR while incrementing IP")
+	}
+
+	return nextIP.String(), nil
+}
+
+// Increments ip k times.
+func incrementIPByValue(ip string, network string, val int) (string, error) {
+	var err error
+	var resIP string
+
+	if strings.Contains(ip, "/") {
+		resIP = strings.Split(ip, "/")[0]
+	} else {
+		resIP = ip
+	}
+
+	for i := 0; i < val; i++ {
+		resIP, err = incrementIP(resIP, network)
+		if err != nil {
+			return "", fmt.Errorf("Failed to increment ip %s: %s", ip, err)
+		}
+	}
+	return resIP, nil
+}
+
+// Get a range of ip addresses.
+// ipsRange[0] is the first ip in the range.
+// ipsRange[1] is the last ip in the range.
+func getIPsRange(firstIP string, network string, rangeSize int) ([]string, error) {
+	ipsRange := make([]string, 2)
+	ipsRange[0] = firstIP
+	lastIP, err := incrementIPByValue(firstIP, network, rangeSize)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to get ips range: %s", err)
+	}
+	ipsRange[1] = lastIP
+
+	return ipsRange, nil
+}
+
+// Compare two ip addresses.
+func compareIPs(ipA string, ipB string) int {
+	maskA := ""
+	maskB := ""
+	splitA := strings.Split(ipA, "/")
+	addressA := splitA[0]
+	if len(splitA) == 2 {
+		maskA = splitA[1]
+	}
+	splitB := strings.Split(ipB, "/")
+	addressB := splitB[0]
+	if len(splitB) == 2 {
+		maskB = splitB[1]
+	}
+	ipCmp := bytes.Compare(net.ParseIP(addressA), net.ParseIP(addressB))
+	if ipCmp < 0 {
+		return -1
+	} else if ipCmp > 0 {
+		return 1
+	} else {
+		mA, _ := strconv.Atoi(maskA)
+		mB, _ := strconv.Atoi(maskB)
+		if mA > mB {
+			return 1
+		} else if mA < mB {
+			return -1
+		} else {
+			return 0
+		}
+	}
+}
+
+// Sort a List of ip addresses.
+func sortIPs(ips []string) []string {
+	sort.Slice(ips, func(i, j int) bool {
+		return compareIPs(ips[i], ips[j]) < 0
+	})
+
+	return ips
+}
+
+// Copy all the files in src directory to the dest directory.
+func CopyDir(src string, dest string) error {
```

Comment:
Could we just do an os call to `cp` or is the a go library that already does this?

Reply:
i need to copy all the files from one directory to another so i use the read and write of "io/ioutil" go library. is there a better way?

Reply:
+1